src/commands/hop.tsx

```diff
@@ -1,51 +1,76 @@
-import { Box, Text, useApp } from 'ink';
+import { Box, Text } from 'ink';
+import { CommandProps } from '../command-registry.js';
 import { useGit } from '../hooks/use-git.js';
 
-function Hop() {
+function Hop({ input }: CommandProps) {
     const git = useGit();
-    const { exit } = useApp();
     const [allBranches, setAllBranches] = useState<string[]>([]);
-    const [currentBranch, setCurrentBranch] = useState('');
+    const [currentBranch, setCurrentBranch] = useState<string | undefined>(
+        undefined
+    );
     const [newBranch, setNewBranch] = useState<string | undefined>(undefined);
     const [error, setError] = useState<Error | undefined>(undefined);
 
+    const [maybeSearchTerm] = input;
+
+    useEffect(() => {
+        if (currentBranch) return;
+        const getLocalBranches = async () => {
+            const { current } = await git.branchLocal();
+            setCurrentBranch((prev) => prev ?? current);
+        };
+        void getLocalBranches();
+    }, [git, setCurrentBranch, currentBranch]);
+
     useEffect(() => {
         const getLocalBranches = async () => {
-            const { all, current } = await git.branchLocal();
+            const { all } = await git.branchLocal();
             setAllBranches(all);
-            setCurrentBranch(current);
         };
 
         void getLocalBranches();
-    }, [git, setAllBranches, setCurrentBranch]);
+    }, [git, setAllBranches]);
 
-    const handleSelect = (item: { label: string; value: string }) => {
-        const updateCurrentBranch = async () => {
-            const { current } = await git.branchLocal();
-            setNewBranch(current);
-        };
+    const handleSelect = useCallback(
+        (item: { label: string; value: string }) => {
+            const updateCurrentBranch = async () => {
+                const { current } = await git.branchLocal();
+                setNewBranch(current);
+            };
 
-        git.checkout(item.value)
-            .then(async () => {
-                return updateCurrentBranch();
-            })
-            .catch((error: Error) => {
-                setError(error);
-            })
-            .finally(() => {
-                exit();
-            });
-    };
+            git.checkout(item.value)
+                .then(() => {
+                    void updateCurrentBranch();
+                })
+                .catch((error: Error) => {
+                    setError(error);
+                });
+        },
+        [git, setNewBranch, setError]
+    );
 
     const items = useMemo(() => {
+        if (!currentBranch) return [];
+
         if (allBranches.length === 1) {
             return [{ label: currentBranch, value: currentBranch }];
         }
-        return allBranches
+
+        const searchedBranches = allBranches.filter((branch) =>
+            maybeSearchTerm
+                ? branch.toLowerCase().includes(maybeSearchTerm.toLowerCase())
+                : true
```

Comment:
An empty string being included will always be true for a string, so this could be simplified

Reply:
`maybeSearchTerm` is undefined when not specified